internal/transport/transport_test.go

```diff
@@ -2136,6 +2137,69 @@ func (s) TestHeadersHTTPStatusGRPCStatus(t *testing.T) {
 	}
 }
 
+func (s) TestWriteHeaderConnectionError(t *testing.T) {
+	server, client, cancel := setUp(t, 0, notifyCall)
+	defer cancel()
+	defer server.stop()
+
+	waitWhileTrue(t, func() (bool, error) {
+		server.mu.Lock()
+		defer server.mu.Unlock()
+
+		if len(server.conns) == 0 {
+			return true, fmt.Errorf("timed-out while waiting for connection to be created on the server")
+		}
+		return false, nil
+	})
+
+	if len(server.conns) != 1 {
+		t.Fatal("Server must have an active connection for the client.")
+	}
+
+	// Get the server transfort for the connecton to the client
+	var serverTransport *http2Server
+	server.mu.Lock()
+	for k := range server.conns {
+		serverTransport = k.(*http2Server)
+	}
+	notifyChan := make(chan struct{})
+	server.h.notify = notifyChan
+	server.mu.Unlock()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+	cstream1, err := client.NewStream(ctx, &CallHdr{})
+	if err != nil {
+		t.Fatalf("Client failed to create first stream. Err: %v", err)
+	}
+
+	<-notifyChan // Wait server stream to be established
+	var sstream1 *Stream
+	// Access stream on the server
+	serverTransport.mu.Lock()
+	for _, v := range serverTransport.activeStreams {
+		if v.id == cstream1.id {
+			sstream1 = v
+		}
+	}
+	serverTransport.mu.Unlock()
+	if sstream1 == nil {
+		t.Fatalf("Didn't find stream corresponding to client cstream.id: %v on the server", cstream1.id)
+	}
+
+	client.Close(fmt.Errorf("closed manually by test"))
+
+	// Wait server transport to be closed
+	<-serverTransport.done
+
+	// Write header on a closed server transport
+	err = serverTransport.WriteHeader(sstream1, metadata.MD{})
+	st := status.Convert(err)
+	if st.Code() != codes.Unavailable {
+		t.Fatalf("Unailable status expected but got: %v", st.Code().String())
```

Comment:
Always specify got before want in test failure messages. See: https://google.github.io/styleguide/go/decisions.html#got-before-want

`t.Fatalf("WriteHeader() failed with status code %s, want %s", st.Code(), codes.Unavailable)`

Note that since `codes.Code` implements the `stringer` interface, we can directly use the %s formatting directive without using the `String()` method.

Reply:
Thanks for the explanations. Updated the failure message.